### DISCUSSION TITLE
How to get detailed exception info

### DISCUSSION THREAD

**Discussion by llamafilm** [encode/httpx#2577] [Potential Issue]
What is the recommended way to see detailed connection errors for GET connections?  There are many different exception classes in httpx, and I would handle them all in the same way, but I want to print a detailed error message to the debug log for troubleshooting.  All the exception classes are empty except for a docstring, so this is what I came up with.  But I'm not sure if it's good practice to rely on docstrings like this?

```python
try:
    httpx.get('http://127.0.0.1/badpage')
except httpx.HTTPError as e:
    print(e.__doc__.strip())
```

For a realistic example see https://github.com/llamafilm/iotawattpy/blob/patch-1/iotawattpy/connection.py#L45

**Comment by ZzZombo** [comment]
Thank you for the hint, finally found out why do requests fail sometimes. It's really bizarre why by default stringification of exceptions results in empty strings.

**Comment by ydennisy** [comment]
Yeah I have a similar issue all I get in stacktrace is:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/httpx/_transports/default.py", line 60, in map_httpcore_exceptions
    yield
  File "/usr/local/lib/python3.11/site-packages/httpx/_transports/default.py", line 353, in handle_async_request
    resp = await self._pool.handle_async_request(req) 
```

Any idea how to find where this is being cause and why?
